src/sdk.hpp

```diff
@@ -0,0 +1,187 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef realm_sdk_h
+#define realm_sdk_h
+
+#include "sdk/object.hpp"
+#include "sdk/optional.hpp"
+#include "sdk/string.hpp"
+#include "sdk/primitive.hpp"
+#include "sdk/arithmetic.hpp"
+#include "sdk/list.hpp"
+#include "sdk/realm.hpp"
+#include "sdk/time_point.hpp"
+#include "sdk/object_id.hpp"
+#include "sdk/property.hpp"
+
+// MARK: Macro Utilities
+
+#define SECOND(a, b, ...) b
```

Comment:
All these macros clobber the global macro namespace for any user. They have some pretty generic names, too.